src/Agent.Worker/TaskManager.cs

```diff
@@ -76,6 +76,8 @@ into taskGrouping
                 return;
             }
 
+            List<Guid> exceptionList = GetTaskExceptionList();
```

Comment:
technically it could be HashSet and then `Contains` would not enumerate

Reply:
Fixed, thanks!